### PR TITLE
Add change password functionality

### PR DESCRIPTION
**Pull request by AllenAnthes**
# Description of changes
Adds a `change_password` page for authenticated users, as well as a link to it from the base `/profile` page

Also refactored the `PasswordResetForm` component to be more generic, working for both the reset password and change password requests.

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 26ad1d2bdee1a9639a797aaefddf47432776527d

https://deploy-preview-563--operation-code-storybook.netlify.com

File: components/ChangePasswordForm/ChangePasswordForm.js [hunk 1]
```diff
@@ -22,9 +22,13 @@ const passwordResetSubmitSchema = Yup.object().shape({
     .oneOf([Yup.ref('newPassword1')], validationErrorMessages.passwordMatch),
 });
 
-export default class PasswordResetSubmitForm extends React.Component {
+/**
+ * Form component used for changing a password either during a password reset
+ * or standard change password request.
+ */
+export default class ChangePasswordForm extends React.Component {
```
Comment (kylemh) [inline components/ChangePasswordForm/ChangePasswordForm.js]: nice abstraction

File: pages/styles/profile.css [hunk 2]
```diff
@@ -1,4 +1,6 @@
 .actionItems {
+  display: flex;
+  justify-content: space-evenly;
```
Comment (kylemh) [inline pages/styles/profile.css]: https://caniuse.com/#search=space-evenly

how about `space-around`

Comment (AllenAnthes) [reply pages/styles/profile.css]: That'll work

File: cypress/integration/profile/change_password.spec.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    cy.get('a[href="/profile"]').should('contain', 'Click here to return to Profile');
+  });
+
+  it('should NOT be able to change with mis-matched passwords', () => {
+    cy.get('#newPassword1').type(existingUser.password);
+    cy.get('#newPassword2').type(`${existingUser.password}1`);
+    cy.get('button[type="submit"]').click();
+
+    cy.get('div[role="alert"]').should('contain', validationErrorMessages.passwordMatch);
+  });
```
Comment (kylemh) [inline cypress/integration/profile/change_password.spec.js]: What happens if the server is down?

File: pages/profile/change_password.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import withAuthSync from 'decorators/withAuthSync/withAuthSync';
+import { changePassword } from 'common/constants/api';
+import Head from 'components/head';
+import HeroBanner from 'components/HeroBanner/HeroBanner';
+import Content from 'components/Content/Content';
+import ChangePasswordForm from 'components/ChangePasswordForm/ChangePasswordForm';
+import styles from '../styles/password_reset.css';
+
+class ChangePassword extends React.Component {
+  static propTypes = {
```
Comment (kylemh) [inline pages/profile/change_password.js]: how would props get passed thru to these pages?

Comment (AllenAnthes) [reply pages/profile/change_password.js]: Whoops, caught me copy-pasting

File: cypress/integration/profile/change_password.spec.js (hunk 3 above)
Comment (AllenAnthes) [reply cypress/integration/profile/change_password.spec.js]: The standard `Something is wrong on our end. Please try again later.` message.  Want me to add a test for it?

Comment (kylemh) [reply cypress/integration/profile/change_password.spec.js]: Yeah. Sorry that’s what I was getting at

**Pull request merged**